app/views/teachers/groups/_form.html.erb

```diff
@@ -13,30 +13,30 @@
 
   <div>
     <%= form.label :name, style: "display: block" %>
-    <%= form.text_field :name %>
+    <%= form.text_field :name, value: group.name %>
   </div>
 
   <div>
     <%= form.label :from_time, style: "display: block" %>
-    <%= form.time_field :from_time %>
+    <%= form.time_field :from_time, value: group.from_time&.strftime('%T') %>
   </div>
 
   <div>
     <%= form.label :to_time, style: "display: block" %>
-    <%= form.time_field :to_time %>
+    <%= form.time_field :to_time, value: group.to_time&.strftime('%T') %>
   </div>
 
   <div>
     <%= form.label :days_of_the_week, style: "display: block" %>
     
     <%= form.fields_for :days_of_the_week do |field| %>
-      sunday: <%= field.check_box :sunday? %>
-      monday: <%= field.check_box :monday? %>
-      tuesday: <%= field.check_box :tuesday? %>
-      wednesday: <%= field.check_box :wednesday? %>
-      thursday: <%= field.check_box :thursday? %>
-      friday: <%= field.check_box :friday? %>
-      saturday: <%= field.check_box :saturday? %>
+      sunday: <%= field.check_box :sunday, { checked: group[:days_of_the_week][:sunday] == "1" ? true : false } %>
```

Comment:
take a look at how you use `group[:days_of_the_week][:sunday] == "1"`, do you really need a ternary operator?

Reply:
You're right. Fixed.